spring-cloud-commons/src/main/java/org/springframework/cloud/client/hedger/HedgingClient.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.client.hedger;
+
+import java.time.Duration;
+
+import org.springframework.lang.Nullable;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ClientResponse;
+
+/**
+ * @author Kevin Binswanger
+ */
+public interface HedgingClient {
+	boolean shouldHedge(ClientRequest request);
```

Comment:
It might make sense to combine these 3 methods into something like:
`ClientRequestHedgeConfiguration getHedgeConfiguration(ClientRequest request)`

Reply:
That requires making this extra object, the configuration. I'll think about it because it makes the ExchangeFilterFunction a slight bit simpler.

Reply:
I do think we could probably get rid of `shouldHedge` and replace it with `getNumberOfHedgedRequests` returning 0.